### PR TITLE
Closes #422 cli messaging update for `assert_date_var()`

### LINKED ISSUE #422: `assert_date_var()` (depends on `assert_symbol()`, `assert_data_frame()`)

### PR DESCRIPTION
**Pull request by ddsjoberg**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiral/dev/CONTRIBUTING.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `main` branch until you have checked off each task.

- [x] Place Closes #422 into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#deprecation)?
- [x] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` under the header `# admiral (development version)` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiraldev/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

File: R/assertions.R [hunk 1]
```diff
@@ -1698,27 +1700,29 @@ assert_one_to_one <- function(dataset, vars1, vars2) {
...
-    abort(paste0(
-      "`",
-      var_name,
-      "` in dataset `",
-      dataset_name,
-      "` is not a date or datetime variable but is ",
-      friendly_type_of(column)
-    ))
+    message <- message %||%
+      "Column {.val {var_name}} in data frame {.code {dataset_name}} must be
```
Comment (bms63) [inline R/assertions.R:1718]: Is there a way to get the name of the dataset object for the user? 
image

Comment (ddsjoberg) [reply R/assertions.R:1718]: @bms63 probably? I just implemented the same messaging in the previous iteration. I'll see what i can do

File: R/assertions.R [hunk 2]
```diff
@@ -1698,27 +1700,29 @@ assert_one_to_one <- function(dataset, vars1, vars2) {
...
+                            dataset_name = rlang::caller_arg(dataset),
+                            var_name = rlang::caller_arg(var),
+                            message = NULL,
+                            class = "assert_date_var",
+                            call = parent.frame()) {
   var <- assert_symbol(enexpr(var))
   assert_data_frame(dataset, required_vars = exprs(!!var))
-  assert_character_scalar(dataset_name, optional = TRUE)
-  assert_character_scalar(var_name, optional = TRUE)
+  assert_character_scalar(dataset_name)
```
Comment (bms63) [inline R/assertions.R:1712]: I think removing the optional = TRUE is good, but wondering why we had it that way?  @bundfussr @zdz2101 do you remember?

**Review by bms63 — COMMENTED** [review]
Looks good to me!  Could someone else take a look as well please

File: R/assertions.R (hunk 2 above)
Comment (ddsjoberg) [reply R/assertions.R:1712]: In the previous iteration the default for the `dataset_name` argument was `NULL`, and we checked that is was indeed a string if a value was passed. In this update, we have a default value that should be a string.

Comment (ddsjoberg) [reply R/assertions.R:1712]: @bms63 I think your request makes perfect sense in a simple situation like `mtcars |> foo()`: you would like to see `mtcars`.

But we can quickly get into some ambiguous situations:

```r
mtcars %>% foo()                        # here the data frame is named `.`
mtcars %>% filter(cyl == 6) %>% foo()   # the data frame is again named `.`
mtcars |> filter(dyl == 6) |> foo()     # here the data frame is named `filter(mtcars, cyl == 6)`
```

What do you think? Should we just stick with the argument name? Then it's always defined.

File: R/assertions.R (hunk 1 above)
Comment (ddsjoberg) [reply R/assertions.R:1718]: @bms63 I think your request makes perfect sense in a simple situation like `mtcars |> foo()`: you would like to see `mtcars`.

But we can quickly get into some ambiguous situations:

```r
mtcars %>% foo()                        # here the data frame is named `.`
mtcars %>% filter(cyl == 6) %>% foo()   # the data frame is again named `.`
mtcars |> filter(cyl == 6) |> foo()     # here the data frame is named `filter(mtcars, cyl == 6)`
```

What do you think? Should we just stick with the argument name? Then it's always defined.

Comment (bms63) [reply R/assertions.R:1718]: Ah - yes.  keep it simple!

File: R/assertions.R [hunk 3]
```diff
@@ -1714,27 +1716,29 @@ assert_one_to_one <- function(dataset, vars1, vars2) {
...
-    abort(paste0(
-      "`",
-      var_name,
-      "` in dataset `",
-      dataset_name,
-      "` is not a date or datetime variable but is ",
-      friendly_type_of(column)
-    ))
+    message <- message %||%
+      "Column {.val {var_name}} in data frame {.code {dataset_name}} must be
```
Comment (bundfussr) [inline R/assertions.R:1734]:
```suggestion
      "Column {.val {var_name}} in dataset {.code {dataset_name}} must be
```
To be consistent with documentation and other messages?

File: R/assertions.R [hunk 4]
```diff
@@ -1665,6 +1665,8 @@ assert_one_to_one <- function(dataset, vars1, vars2) {
 #' @param var_name The name of the variable. If the argument is specified, the
 #'   specified name is displayed in the error message.
 #'
+#' @inheritParams assert_logical_scalar
```
Comment (bundfussr) [inline R/assertions.R:1668]: We can't take the documentation of `message` from `assert_logical_scalar()` because there is no `arg_name` argument.
image

Comment (ddsjoberg) [reply R/assertions.R:1671]: good catch, thank you!. updated!

**Review by bundfussr — APPROVED** [review]

**Review by bms63 — APPROVED** [review]
